source/client-side-operations-timeout/etc/templates/override-collection-timeoutMS.yml.template

```diff
@@ -0,0 +1,135 @@
+# Tests in this file are generated from override-collection-timeoutMS.yml.template.
+
+description: "timeoutMS can be overriden for a MongoCollection"
+
+schemaVersion: "1.2"
+
+runOnRequirements:
+  - minServerVersion: "4.4"
+
+createEntities:
+  - client:
+      id: &failPointClient failPointClient
+      useMultipleMongoses: false
+  - client:
+      id: &client client
+      uriOptions:
+        timeoutMS: 10
+      useMultipleMongoses: false
+      observeEvents:
+        - commandStartedEvent
+      ignoreCommandMonitoringEvents:
+        - killCursors
+  - database:
+      id: &database database
+      client: *client
+      databaseName: &databaseName test
+
+initialData:
+  - collectionName: &collectionName coll
+    databaseName: *databaseName
+    documents: []
+
+tests:
+  # For each collection-level operation, we execute two tests:
+  #
+  # 1. timeoutMS can be overridden to a non-zero value for a MongoCollection. Each test constructs uses the client
+  # entity defined above to construct a collection entity with timeoutMS=1000 and configures a fail point to block the
+  # operation for 15ms so the operation succeeds.
```

Comment:
If we don't set a fail point, the test would pass even if a driver had a bug that caused the collection timeoutMS to be 10 (inherited from the client) because the operation wouldn't time out. By configuring the operation to block for longer than the client's timeoutMS, we're making sure that the collection's timeoutMS was correctly overridden to infinite.